src/state/Data/actions.js

```diff
@@ -60,6 +65,84 @@ const getRestRelationsPages = (
 	}
 };
 
+// Check if all of spatialDataSources are tiled type.
+const checkAllDSAreTiled = spatialDataSources => {
+	const DSTILED = !_isEmpty(spatialDataSources)
+		? spatialDataSources.every(ds =>
+				TILED_VECTOR_LAYER_TYPES.includes(ds.data?.type)
+		  )
+		: false;
+	return DSTILED;
+};
+
+const checkSomeDSAreVector = spatialDataSources => {
+	const SDSvectorType = 'vector';
+	const someDSisVectorType = !_isEmpty(spatialDataSources)
+		? spatialDataSources.some(ds => ds.data?.type === SDSvectorType)
+		: false;
+	return someDSisVectorType;
+};
+
+const transformSDStoTypes = spatialDataSources => {
```

Comment:
Why is the transformation needed?

Reply:
It creates objects just with key and type. These objects are used for determining if all or some of spatialDataSources are type of vector/raster etc.

So motivation for transformation is simplification of complicated objects.